app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::V1::CommentsController < ApplicationController
+  before_action :set_user_and_post, only: %i[create]
+
+  load_and_authorize_resource
+
+  def index
+    @comments = Comment.where(post_id: params[:post_id], author_id: params[:user_id])
+    render json: @comments
+  end
+
+  def create
+    @comment = @post.comments.build(comment_params.merge(author_id: @user.id))
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+
+  def set_user_and_post
+    return if request.headers['Authorization'].nil?
+
+    token = request.headers['Authorization'].split.last
+    @user = User.find_by(token:)
+    @post = @user.posts.includes(:comments).find(params[:post_id])
+  end
```

Comment:
- [ ] I actually commend your going extra mile for adding token as part of the header, howbeit I was unable to post/get comments from the api despite that I added the token generated from the UI. Please look through the screenshot below that I got from the terminal. 🙏 

<img width="440" alt="Untitled" src="https://github.com/Chimoneg27/Rails-Blog-App/assets/103386183/3ea1a605-1fa2-45c4-8032-994079ea7495">

- [ ] I got the error message screenshot 👇 when I tried to get a comment from the UI.

<img width="815" alt="Untitled" src="https://github.com/Chimoneg27/Rails-Blog-App/assets/103386183/5faa7d5a-c7cd-4b8d-bedc-39471b2cf929">

Reply:
@OyePriscilla what should I change

Reply:
@Chimoneg27 Perhaps you can turn off the CSRF token validation off; I think this [link](https://stackoverflow.com/questions/35181340/rails-cant-verify-csrf-token-authenticity-when-making-a-post-request) can be of help.